infrastructures/infrastructure-azure/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AzureInfrastructure.java

```diff
@@ -0,0 +1,376 @@
+
+        if (parameters[12] == null) {
+            throw new IllegalArgumentException("The virtual machine username must be specified");
+        }
+
+        if (parameters[13] == null) {
+            throw new IllegalArgumentException("The virtual machine password must be specified");
+        }
+
+        if (parameters[17] == null) {
+            throw new IllegalArgumentException("The number of instances to create must be specified");
+        }
+
+        if (parameters[18] == null) {
+            throw new IllegalArgumentException("The number of nodes per instance to deploy must be specified");
+        }
+
+        if (parameters[19] == null) {
+            throw new IllegalArgumentException("The download node.jar command must be specified");
+        }
+
+        if (parameters[22] == null) {
+            parameters[22] = "";
+        }
+    }
+
+    @Override
+    public void acquireNode() {
+
+        connectorIaasController.waitForConnectorIaasToBeUP();
+
+        connectorIaasController.createAzureInfrastructure(getInfrastructureId(),
+                                                          clientId,
+                                                          secret,
+                                                          domain,
+                                                          subscriptionId,
+                                                          authenticationEndpoint,
+                                                          managementEndpoint,
+                                                          resourceManagerEndpoint,
+                                                          graphEndpoint,
+                                                          false);
+
+        String instanceTag = getInfrastructureId();
+        Set<String> instancesIds;
+        instancesIds = connectorIaasController.createAzureInstances(getInfrastructureId(),
+                                                                    instanceTag,
+                                                                    image,
+                                                                    numberOfInstances,
+                                                                    vmUsername,
+                                                                    vmPassword,
+                                                                    vmPublicKey,
+                                                                    vmSizeType,
+                                                                    resourceGroup,
+                                                                    region,
+                                                                    privateNetworkCIDR,
+                                                                    staticPublicIP);
+
+        logger.info("Instances ids created : " + instancesIds);
+
+        /*
+         * TODO: could be injected on boot just like OpenStack BUT scripts need to be customized
```

Comment:
removed in 06b250351a3cac447f5c0b0272c314c5bd90a0cb
Issue created in https://github.com/ow2-proactive/nodesource-addons/issues/34